release.js

```diff
@@ -374,46 +376,107 @@ function createWebsiteDirectory() {
     if (err && err.code !== 'EEXIST')
-        return pauseRetry(err);
+      return pauseRetry(err);
+    if (fs.readdirSync(distdir).length !== 0)
+      return pauseRetry("dist dir is not empty");
 
     next();
   });
 }
 
 
-function createTarBall() {
+function createCMakeTarBall() {
   var tag = ver.format(state.releaseVersion);
   var filename = format('libuv-%s.tar.gz', tag);
   var prefix = format('libuv-%s/', tag);
   var command = format('git archive %s --format=tar --prefix=%s | gzip -9 > dist/%s',
                        tag,
                        prefix,
                        filename);
-  child_process.exec(command, { stdio: 'inherit', cwd: dir}, nextOrRetry);
+  child_process.exec(command, { stdio: 'inherit', cwd: dir }, nextOrRetry);
 }
 
 
-function signTarball() {
+function createAutotoolsTarBall() {
   var tag = ver.format(state.releaseVersion);
-  var filename = format('libuv-%s.tar.gz', tag);
-  var signFilename = format('libuv-%s.tar.gz.sign', tag);
-  var command = format('gpg -a --output dist/%s --detach-sign dist/%s',
-                       signFilename,
-                       filename);
-  child_process.exec(command, { stdio: 'inherit', cwd: dir}, nextOrRetry);
+  try {
+  var distver = format('%d.%d.%d',
+                       state.releaseVersion.major,
+                       state.releaseVersion.minor,
+                       state.releaseVersion.patch);
+  } catch (err) {
+    return pauseRetry(err);
+  }
+  child_process.execFile('./autogen.sh', [], { stdio: 'inherit', cwd: dir},
+                         function(err, stdout, stderr) {
+    if (err) {
+      console.log(stdout);
+      console.log(stderr);
+      return pauseRetry(err);
+    }
+    try {
+      var builddir = fs.mkdtempSync(path.join(os.tmpdir(), 'libuv-build-'));
+      console.log('    attempting build in %s', builddir);
+    } catch (err) {
+      return pauseRetry(err);
+    }
+    child_process.execFile(dir + '/configure', [],
+                           { stdio: 'inherit', cwd: builddir},
+                           function(err, stdout, stderr) {
+      if (err) {
+        console.log(stdout);
+        console.log(stderr);
+        return pauseRetry(err);
+      }
+      child_process.execFile('make', ['dist', '-j' + os.cpus().length], // TODO: use distcheck here
+                             { stdio: 'pipe', cwd: builddir },
+                             function(err, stdout, stderr) {
+        if (err) {
+          console.log(stdout);
+          console.log(stderr);
+          return pauseRetry(err);
+        }
+        var srcfilename = path.join(builddir, format('libuv-%s.tar.gz', distver));
+        var dstfilename = path.join(dir, 'dist', format('libuv-%s-dist.tar.gz', tag));
+        try {
+          fs.renameSync(srcfilename, dstfilename);
+          fs.rmdirSync(builddir, { recursive: true });
+        } catch (err) {
+          return pauseRetry(err);
+        }
+        next();
+      });
+    });
+  });
+}
```

Comment:
Bit of a callback pyramid of doom going on in this function. I don't think there's any harm in using `.execFileSync()` or async/await.

If you want to stick with the callback style, consider breaking out the steps into separate functions (can be lexically scoped to the outer function.)

Reply:
AFAICT, execFileSync doesn't support stdio access and async/await would require structural changes that I don't want to do right now. I could Promise-ify the code instead, but that seemed like a significant change also, and seems like it forces us to resolve all of them (so we can't pause part way through the steps, but are forced to finish them, even after failure), but I may not understand Promises that well yet. The problem with separate functions is that it forces the code to go in inverse order, which seemed just as bad as the code pyramid formatting issue.